### PR TITLE
Implement Stage::RegisterChain and Stage::AddedCallback

### PR DESCRIPTION
**Pull request by Renzo904**

### CODE REVIEW

File: src/ChainPriorities.hpp [hunk 1]
```diff
@@ -4,9 +4,12 @@
 #define TH_CHAIN_PRIO_CALC_ASCIIMANAGER 1
 #define TH_CHAIN_PRIO_CALC_MAINMENU 2
 #define TH_CHAIN_PRIO_CALC_GAMEMANAGER 4
+#define TH_CHAIN_PRIO_CALC_STAGE 6
 
 #define TH_CHAIN_PRIO_DRAW_MAINMENU 0
 #define TH_CHAIN_PRIO_DRAW_GAMEMANAGER 2
+#define TH_CHAIN_PRIO_DRAW_HIGH_PRIO_STAGE 3
```
Comment (Renzo904) [inline src/ChainPriorities.hpp:11]: Not sure about this name tho, idk if it should be like that or `TH_CHAIN_HIGH_PRIO_DRAW_STAGE`

Comment (roblabla) [reply src/ChainPriorities.hpp:11]: I'd rather not alter the TH_CHAIN_PRIO prefix. This is fine as-is :)

File: src/Colors.hpp [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+/* COLORS */
+#define COLOR_BLACK 0xff000000
```
Comment (roblabla) [inline src/Colors.hpp:4]: We have a ZunColor.hpp file, maybe we should move those in there?

File: src/Stage.cpp [hunk 3]
```diff
@@ -1,3 +1,106 @@
...
+    stage->facingDirInterpInitial = interpInitial;
+
+    stage->facingDirInterpDuration = 1;
+    facingDirTimer = &stage->facingDirInterpTimer;
+    facingDirTimer->current = 0;
+    facingDirTimer->subFrame = 0.0;
+    facingDirTimer->previous = -999;
+    stage->unpauseFlag = 0;
+
+    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGCOLOR, (stage->skyFog).color);
```
Comment (roblabla) [inline src/Stage.cpp:100]:
```suggestion
    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGCOLOR, stage->skyFog.color);
```

File: src/Stage.cpp [hunk 4]
```diff
@@ -1,3 +1,106 @@
...
+    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGSTART, *(DWORD *)&(stage->skyFog).nearPlane);
```
Comment (roblabla) [inline src/Stage.cpp:101]:
```suggestion
    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGSTART, *(DWORD *)&stage->skyFog.nearPlane);
```

File: src/Stage.cpp [hunk 5]
```diff
@@ -1,3 +1,106 @@
...
+    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGEND, *(DWORD *)&(stage->skyFog).farPlane);
```
Comment (roblabla) [inline src/Stage.cpp:102]:
```suggestion
    g_Supervisor.d3dDevice->SetRenderState(D3DRS_FOGEND, *(DWORD *)&stage->skyFog.farPlane);
```

File: src/Stage.cpp [hunk 6]
```diff
@@ -1,3 +1,106 @@
...
+    stage->facingDirInterpFinal = interpFinal;
+
+    interpInitial.x = 0;
+    interpInitial.y = 0;
+    interpInitial.z = 1.0;
+    stage->facingDirInterpInitial = interpInitial;
+
+    stage->facingDirInterpDuration = 1;
+    facingDirTimer = &stage->facingDirInterpTimer;
+    facingDirTimer->current = 0;
```
Comment (roblabla) [inline src/Stage.cpp:95]:
```suggestion
    facingDirTimer->InitializeForPopup();
```

We need to rename this function, it's used all over the place.

File: src/Stage.cpp [hunk 7]
```diff
@@ -1,3 +1,106 @@
...
+{
+
+    Stage *stg = &g_Stage;
+    ZunTimer *timer;
+
+    memset(stg, 0, sizeof(Stage));
+    stg->stdData = NULL;
+
+    timer = &stg->timer;
+    timer->current = 0;
```
Comment (roblabla) [inline src/Stage.cpp:26]:
```suggestion
    timer->InitializeForPopup();
```

**Pull request merged**